libraries/networking/src/udt/PacketList.h

```diff
@@ -27,26 +28,44 @@ class Packet;
 class PacketList : public QIODevice {
     Q_OBJECT
 public:
-    PacketList(PacketType packetType, QByteArray extendedHeader = QByteArray());
+    PacketList(PacketType packetType, QByteArray extendedHeader = QByteArray(), bool isReliable = false, bool isOrdered = false);
+    PacketList(PacketList&& other);
+
+    static std::unique_ptr<PacketList> fromReceivedPackets(std::list<std::unique_ptr<Packet>>&& packets);
     
     virtual bool isSequential() const { return true; }
+
+    bool isReliable() const { return _isReliable; }
+    bool isOrdered() const { return _isOrdered; }
     
     void startSegment();
     void endSegment();
     
     PacketType getType() const { return _packetType; }
     int getNumPackets() const { return _packets.size() + (_currentPacket ? 1 : 0); }
+
+    QByteArray getExtendedHeader() const { return _extendedHeader; }
+
+    size_t getDataSize() const;
     
     void closeCurrentPacket(bool shouldSendEmpty = false);
-    
+
+    QByteArray getAllData();
+
     template<typename T> qint64 readPrimitive(T* data);
     template<typename T> qint64 writePrimitive(const T& data);
+    std::list<std::unique_ptr<Packet>> _packets;
 protected:
     virtual qint64 writeData(const char* data, qint64 maxSize);
     virtual qint64 readData(char* data, qint64 maxSize) { return 0; }
+    PacketType _packetType;
+
     
 private:
     friend class ::LimitedNodeList;
+    friend class Socket;
+    friend class SendQueue;
+    friend class NLPacketList;
```

Comment:
The issue was accessing PacketList object data from a separate NLPacketList instance, so unfortunately changing the access to protected didn't help. I actually had an issue with this and made _packets public, which I need to fix.